docs/asciidoc/static/breaking-changes.asciidoc

```diff
@@ -0,0 +1,30 @@
+[[breaking-changes]]
+== Breaking changes
+
+Version 2.0 of Logstash has some changes that are incompatible with previous versions of Logstash. This section discusses 
+what you need to be aware of when migrating to this version.
+
+[float]
+=== Elasticsearch Output Default
+
+Starting with the 2.0 release of Logstash, the default Logstash output for Elasticsearch is HTTP. To use the `node` or
+`transport` protocols, download the {logstash}plugins-outputs-elasticsearch_java.html[Elasticsearch Java plugin]. The 
+Logstash HTTP output to Elasticsearch now supports sniffing.
+
+==== Configuration Changes
+
+The Elasticsearch output plugin configuration has the following changes:
+
+* The `host` configuration option is now `hosts`
+* New options: `bind_host`, `bind_port`, `cluster`, `embedded`, `embedded_http_port`, `port`, `sniffing_delay
+* The `max_inflight_requests` option, which was deprecated in the 1.5 release, is now removed
+* The `node_name` and `protocol` options have been moved to the `elasticsearch_java` plugin
+
+[float]
+=== Kafka Output Configuration Changes
+
+The 2.0 release of Logstash includes a new version of the Kafka output plugin with significant configuration changes.
+Please compare the documentation pages for the 
+https://www.elastic.co/guide/en/logstash/1.5/plugins-outputs-kafka.html[Logstash 1.5] and
+https://www.elastic.co/guide/en/logstash/2.0/plugins-outputs-kafka.html[Logstash 2.0] versions of the Kafka output plugin 
+and update your configuration files accordingly.
```

Comment:
I though about adding that, but it doesn't seem like a breaking change, so
I decided to reserve it for the release notes. Should it go here instead?

On Wed, Oct 14, 2015 at 7:43 PM, Suyog Rao <notifications@github.com> wrote:

> In docs/asciidoc/static/breaking-changes.asciidoc
> <https://github.com/elastic/logstash/pull/4039#discussion_r42078308>:
>
> > +
> > +The Elasticsearch output plugin configuration has the following changes:
> > +
> > +* The `host` configuration option is now `hosts`
> > +* New options: `bind_host`, `bind_port`, `cluster`, `embedded`, `embedded_http_port`, `port`, `sniffing_delay
> > +* The `max_inflight_requests` option, which was deprecated in the 1.5 release, is now removed
> > +* The `node_name` and `protocol` options have been moved to the `elasticsearch_java` plugin
> > +
> > +[float]
> > +=== Kafka Output Configuration Changes
> > +
> > +The 2.0 release of Logstash includes a new version of the Kafka output plugin with significant configuration changes.
> > +Please compare the documentation pages for the
> > +https://www.elastic.co/guide/en/logstash/1.5/plugins-outputs-kafka.html[Logstash 1.5] and
> > +https://www.elastic.co/guide/en/logstash/2.0/plugins-outputs-kafka.html[Logstash 2.0] versions of the Kafka output plugin
> > +and update your configuration files accordingly.
>
> Also need a breaking change section on default workers.
>
> Now, out of the box, the default value of the filter_workers setting will
> be set to half of the CPU cores of the machine. Increasing the workers
> provides parallelism in filter execution which is crucial when doing
> heavier processing like complex grok patterns or the useragent filter.
>
> Note that, as before, this setting can be changed with the w flag, e.g. bin/logstash
> -f test.config -w 1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/4039/files#r42078308>.
>